cmd/thanos/store.go

```diff
@@ -381,14 +386,25 @@ func runStore(
 
 			level.Info(logger).Log("msg", "initializing bucket store")
 			begin := time.Now()
-			if err := bs.InitialSync(ctx); err != nil {
+
+			//This will stop retrying after 15 seconds.
+			initialSyncCtx, cancel := context.WithTimeout(context.Background(), timeoutDuration*time.Second)
+			defer cancel()
+
+			//If error occurs, it will re-try after every 5 seconds, but only for 15 seconds, (so total re-try is three).
```

Comment:
Same goes for here, hard-coded value in comments are unnecessary. Also the retry function is self-explanatory, so simply comment:
```Retry in case of error.```
Second note: It's retry not re-try :)